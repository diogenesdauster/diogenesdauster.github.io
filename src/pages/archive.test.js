import React from 'react'
import renderer from 'react-test-renderer'
import Archive from './archive.js'

describe('Archive', () =>
  it('renders correctly', () => {
    const data = {
      allMarkdownRemark: {
        edges: [
          {
            node: {
              fields: {
                slug: '/2013/12/28/introducing-hyde',
              },
              frontmatter: {
                title: 'Introducing Hyde',
                postdate: 'December 28th, 2013',
                postyear: '2013',
              },
            },
          },
          {
            node: {
              fields: {
                slug: '/2012/02/07/example-content',
              },
              frontmatter: {
                title: 'Example content',
                postdate: 'February 7th, 2012',
                postyear: '2012',
              },
            },
          },
          {
            node: {
              fields: {
                slug: '/2012/02/06/whats-jekyll',
              },
              frontmatter: {
                title: "What's Jekyll?",
                postdate: 'February 6th, 2012',
                postyear: '2012',
              },
            },
          },
        ],
      },
    }

    const tree = renderer
      .create(<Archive data={data} layoutData={sideData} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  }))
