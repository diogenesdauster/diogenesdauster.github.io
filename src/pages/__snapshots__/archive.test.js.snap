// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Archive renders correctly 1`] = `
Array [
  <div
    className="sidebar"
  >
    <div
      className="container sidebar-sticky"
    >
      <div
        className="sidebar-about"
      >
        <h1>
          <a
            href="/"
          >
            <span
              aria-label="emoji"
              role="img"
            >
              üë®üèª‚Äçüíª
               
            </span>
            <span>
              &lt;/
            </span>
            Di√≥genes Dauster
            <span>
              &gt;
            </span>
          </a>
        </h1>
        <p
          className="lead"
        >
          Code Has No Accent
        </p>
      </div>
      <nav
        className="sidebar-nav"
      >
        <Link
          className="sidebar-nav-item"
          to="/"
        >
          Home
        </Link>
        <Link
          className="sidebar-nav-item"
          to="about"
        >
          About
        </Link>
        <Link
          className="sidebar-nav-item"
          to="archive"
        >
          Archive
        </Link>
        <Link
          className="sidebar-nav-item"
          to="projects"
        >
          Projects
        </Link>
      </nav>
      <p>
        ¬© 
        2018
        . All rights reserved.
      </p>
    </div>
  </div>,
  <div
    className="content container"
  >
    <div
      className="page"
    >
      <h1
        className="page-title"
      >
        Archives
      </h1>
      <h1>
        2013
      </h1>
      <ul>
        <li>
          <p>
            December 28th, 2013
             ‚Äî
             
            <Link
              to="/2013/12/28/introducing-hyde"
            >
              Introducing Hyde
            </Link>
          </p>
        </li>
      </ul>
      <h1>
        2012
      </h1>
      <ul>
        <li>
          <p>
            February 7th, 2012
             ‚Äî
             
            <Link
              to="/2012/02/07/example-content"
            >
              Example content
            </Link>
          </p>
        </li>
        <li>
          <p>
            February 6th, 2012
             ‚Äî
             
            <Link
              to="/2012/02/06/whats-jekyll"
            >
              What's Jekyll?
            </Link>
          </p>
        </li>
      </ul>
    </div>
  </div>,
]
`;
